### PR TITLE
feat(Select): Implement Select with search and multiple select

### PR DESCRIPTION
**Pull request by HabRonan**
- Select with default, searchable and multiple select.

### CODE REVIEW

**Comment by lukasbals** [comment]
Some feedback on the handling of the `Select`. 

* It would be nice if it closed if the user clicked anywhere else on the screen. Do you think we can implement that behaviour?
* It would be good if when you open the select it's possible to select items either with the arrow keys or by using the tab key. What do you think?

**Comment by ClaraTschamon** [comment]
In the searchable it would be Select in would be cool if I would automatically be in the searchbar when the Select opens.
So that I don't need to klick into the searchbar before typing.

**Comment by lialila** [comment]
LGTM

File: src/components/Select/Select.test.tsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+import { render, screen, fireEvent } from '@testing-library/react';
+import { Select } from '.';
+
+const mockOptions = [
+  { label: 'Option 1', value: '1' },
+  { label: 'Option 2', value: '2' },
+  { label: 'Option 3', value: '3' },
+];
+
+describe('Select', () => {
```
Comment (lukasbals) [inline src/components/Select/Select.test.tsx:11]:
```suggestion
describe('The Select component', () => {
```

File: src/components/Select/Select.tsx [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+    <Box position="relative" width={isFullWidth ? '100%' : 'auto'} ref={selectRef}>
+      <Flex
+        onClick={toggleDropdown}
+        border="1px"
+        borderColor={isInvalid ? 'red' : 'gray.300'}
+        p="2"
+        align="center"
+        cursor={isDisabled ? 'not-allowed' : 'pointer'}
+        bg={isDisabled ? 'gray.200' : 'white'}
+        borderRadius="md"
```
Comment (lukasbals) [inline src/components/Select/Select.tsx:262]: Is border radius medium enough?

**Comment by lukasbals** [comment]
If there is a selected item, e.g. Option 2, and you open the select the one with the grey background is still "Option 1". 

See an example here: https://boemly-git-ha-multiple-select-treely.vercel.app/?path=/story/components-select--with-selected-value

**Comment by lukasbals** [comment]
Please let @tobsut97  do a design review. ok @tobsut97?

File: src/components/Select/Select.tsx (hunk 2 above)
Comment (HabRonan) [reply src/components/Select/Select.tsx:262]: yes, I think medium is .375 rem or 6 px as in the design.

File: src/components/Select/Select.test.tsx (hunk 1 above)
Comment (HabRonan) [reply src/components/Select/Select.test.tsx]: updated.

**Review by rschiffer — COMMENTED** [review]
LGTM and works as expected 🙂 

One thing which I noticed when the `Select` is set to `isMultiple`: I personally, find it not intuative that the last clicked option is always marked grey. 
What do you think? 
This is really just a personal opinion and also not a strong one. So, it's definitely also fine the way it is.

**Comment by ClaraTschamon** [comment]
Looks really good to me. 😍  Well done!

**Review by lukasbals — COMMENTED** [review]
Nice work! 

You need to make sure that this deployment is marked as a breaking change. This means you need to define the commit message accordingly: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit-message-footer

File: src/components/Select/Select.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    onChange: { action: 'Select Changed' },
+    options: { control: 'object' },
+    color: {
+      options: ['black', 'white'],
+      control: { type: 'radio' },
+    },
+    borderColor: {
+      options: ['gray.200', 'white'],
+      control: { type: 'radio' },
+    },
```
Comment (lukasbals) [inline src/components/Select/Select.stories.tsx:24]: I think those are not the only options for colors. So it should probably not be a radio but only a string input field? 

And I think `backgroundColor` is missing here?

Comment (HabRonan) [reply src/components/Select/Select.stories.tsx]: Makes sense, updated.

**Comment by HabRonan** [comment]
- Should we introduce a searchPlaceholder text, different from the select placeholder? so we have different text for the select and the search
- need to check why the BoemlyFormControl isDisabled and isInvalid isn't getting passed to the new Select.

**Comment by lukasbals** [comment]
> Notes:
> 
> * Should we introduce a searchPlaceholder text, different from the select placeholder? so we have different text for the select and the search
> * need to check why the BoemlyFormControl isDisabled and isInvalid isn't getting passed to the new Select.

1. It Would make sense IMO. But could also be done on demand
2. 👍🏽

**Review by rschiffer — APPROVED** [review]
So nice!!!

**Review by lukasbals — APPROVED** [review]
Machine!

**Pull request merged**